packages/kitsu-core/src/query/index.js

```diff
@@ -7,7 +7,8 @@
  * @private
  */
 function queryFormat (value, key) {
-  if (value !== null && typeof value === 'object') return query(value, key)
+  if (value !== null && Array.isArray(value)) return value.map(v => queryFormat(v, `${key}[]`)).join('&')
```

Comment:
I'm unsure if its proper for the library to include `[]` at the end of the key, or if its better to expect the user to include it at the end of the key themself like this `query({filter: {"id_in[]": [1,2,3] }})`

On one hand many backends use `[]` at the end to denote an array value (PHP, Rails and more) so its a pretty sane choice, but on the other hand this makes the library more opinionated.

I'm using Rails, so opinionated is fine with me, but i think the decision should be yours @wopian

Reply:
I think we should go with the unopinionated route to match the JSON:API specs which accepts whatever - so API server libraries in various languages could use any of the formats the specification lists as potentially possible for query parameter filters (`filter[x]`, `filter[x][]` etc.,).

If the API a user of `kitsu`/`kitsu-core` uses the `filter[x][]` format they'll know to include `[]` from that API's own documentation. 👍

Reply:
https://jsonapi.org/format/#query-parameters-families

> For example, the filter query parameter family includes parameters named: filter, filter[x], filter[], filter[x][], filter[][], filter[x][y]